Firestore/core/test/unit/testutil/testutil.cc

```diff
@@ -648,6 +648,21 @@ model::FieldIndex MakeFieldIndex(const std::string& collection_group,
           state};
 }
 
+std::unordered_map<model::DocumentKey,
+                   model::Mutation::Type,
+                   model::DocumentKeyHash>
+OverlayTypeMap(
+    std::vector<std::pair<model::DocumentKey, model::Mutation::Type>> pairs) {
+  std::unordered_map<model::DocumentKey, model::Mutation::Type,
+                     model::DocumentKeyHash>
+      result;
+  for (const auto& p : pairs) {
+    result.emplace(p.first, p.second);
+  }
+
+  return result;
```

Comment:
Is this function necessary? I think you should be able to directly initialize the `unordered_map` using the initializer list syntax. See examples here: https://en.cppreference.com/w/cpp/container/unordered_map/unordered_map